da/celestia/celestia.go

```diff
@@ -138,39 +139,55 @@ func (c *DataAvailabilityLayerClient) Start() (err error) {
 		return err
 	}
 
-	state, err := rpc.Header.SyncState(c.ctx)
-	if err != nil {
-		return err
-	}
-
-	if !state.Finished() {
-		c.logger.Info("waiting for celestia-node to finish syncing", "height", state.Height, "target", state.ToHeight)
-
-		done := make(chan error, 1)
-		go func() {
-			done <- rpc.Header.SyncWait(c.ctx)
-		}()
-
-		ticker := time.NewTicker(1 * time.Minute)
-		defer ticker.Stop()
+	go retry.Do(
```

Comment:
I think simple for loop, with `state.Finished()` as break condition is more clearer.
IMO we shouldn't retry on errors for those queries, as it's an init process and errors should be popped up

Reply:
a simple for loop will not start a celestia syncing process in background. the idea of this pr is that only sequencers must wait for syncing and full nodes can start receiving blocks without waiting for it. i think in this case, retry is necessary in case of any error or disconnection with da light happens, to continue with the process, without having to panic.